odoo/tools/translate.py

```diff
@@ -785,7 +785,11 @@ def write_rows(self, rows):
 
 def trans_export(lang, modules, buffer, format, cr):
 
-    translations = trans_generate(lang, modules, cr)
+    reader = TranslationExporter(cr, modules=modules)
+    reader.export_translatable_records()
+    reader.export_translatable_resources()
+    translations = reader.read_rows(lang)
```

Comment:
The `TranslationExporter` just makes the API more confusing from a user's point of view:
- this is not a "reader" in the usual meaning of the term;
- the methods `export_translatable_records` and `export_translatable_resources` look cryptic:
   - should they be public?
   - does it make any sense to not call both of them?

I suggest the following alternative API:
```python
translations = TranslationExporter(cr, modules).get_rows(lang)
```